src/cwhy/cwhy.py

```diff
@@ -118,24 +118,27 @@ def evaluate(args, stdin):
 
 def evaluate_prompt(args, prompt, wrap=True):
     if args["show_prompt"]:
+        import sys
         print("===================== Prompt =====================")
         print(prompt)
         print("==================================================")
+        sys.exit(0)
     text = complete(args, prompt)
     if wrap:
         text = word_wrap_except_code_blocks(text)
     return text
 
 
 class explain_context:
-    def __init__(self, diagnostic):
+    def __init__(self, args, diagnostic):
+        
         diagnostic_lines = diagnostic.splitlines()
 
         self.code_locations = {}
 
-        # Magic number - don't send more than this many code locations.
+        # Don't send more than this many code locations.
         # This is just to prevent overwhelming OpenAI.
-        max_code_locations = 10
+        max_code_locations = int(args["max_context"])
```

Comment:
```suggestion
        max_code_locations = args["max_context"]
```

Nit, this is already an int from argparse's `type=int`.